### ISSUE TITLE
move_and_collide registers CSGshapes with a frame of delay.

### ISSUE THREAD

**Issue by yosoyfreeman** [godotengine/godot#98125]
### Tested versions

-Reproducible in 4.3

### System information

Linux mint 

### Issue description

When the game starts running the body will register no collisions until one frame has passed if the would be collision is against a CSG shape. I'm guessing there is some kind of delay between the CSG shapes spawning and the collision shape being added on the world state.

If the guess is correct and this is inherent to CSG shapes and we can't do nothing without heavy changes, i suggest documenting it, as is in my opinion disruptive enough as to cause some headaches to the user.

### Steps to reproduce

Open the project in godot. There are 2 scenes: csgfloor and staticfloor. When run, staticfloor will report correctly on the first frame. csgfloor will take an extra frame to do so under the same conditions.

### Minimal reproduction project (MRP)

[csgbug.tar.gz](https://github.com/user-attachments/files/17351417/csgbug.tar.gz)

**Comment by smix8** [comment]
This is due to a `CSGShape3D` Node doing deferred `shape` updates as a CSGShape3D result depends on other potential CSG nodes inside the SceneTree.

**Comment by TokageItLab** [comment]
Also related by https://github.com/godotengine/godot/issues/70804#issuecomment-1368888589

**Comment by smix8** [comment]
This is not a CSG specific problem.

Anything that does deferred physics updates has the same issues because the physics loop flushes the main message  queue only after its step, not before its step.
